### ISSUE TITLE
IDL sequence of arrays fails.

### ISSUE THREAD

**Issue by matte1** [eclipse-cyclonedds/cyclonedds#1712]
The following IDL type fails:
`sequence<double[2]> foo;`
but,
```
      typedef double Bar[2];
      struct Data {
        sequence<Bar> foo;
      };
```
is okay?

**Comment by dpotman** [comment]
Yes, I think this is the correct behavior, the IDL 4.1 spec allows array declarators on an `identifier` (e.g. a member of an aggregated type, or a typedef), and not on a `type_spec`. See [https://www.omg.org/spec/IDL/4.2/PDF](https://www.omg.org/spec/IDL/4.2/PDF) section 7.4.1.4.4.5.
